src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -169,6 +170,41 @@ public static Set<Tag> parseTags(Collection<String> tags) throws ParseException
         return tagSet;
     }
 
+    /**
+     * Parses a {@code String medicalHistory} into a {@code medicalHistory}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code tag} is invalid.
+     */
+    public static MedicalHistory parseMedicalHistory(String medicalHistory) throws ParseException {
+        requireNonNull(medicalHistory);
+        String trimmedMedicalHistory = medicalHistory.trim();
+        if (!MedicalHistory.isValidMedicalHistory(medicalHistory)) {
+            throw new ParseException(Tag.MESSAGE_CONSTRAINTS);
```

Comment:
Wrong Constraint used for `MedicalHistory`.

Reply:
My bad.